force_wfmanager/left_side_pane/workflow_tree.py

```diff
@@ -283,16 +334,154 @@ def new_data_source(self, ui_info, object):
 
+    @verify
     def edit_data_source(self, ui_info, object):
         # This is a live dialog, workaround for issue #58
         object.model.edit_traits(kind="livemodal")
 
+    @verify
     def new_layer(self, ui_info, object):
         self.workflow_mv.add_execution_layer(ExecutionLayer())
 
+    @verify
     def delete_layer(self, ui_info, object):
         """ Delete an element from the workflow """
         self.workflow_mv.remove_execution_layer(object.model)
+
+    @on_trait_change("selected_mv,verify_workflow_event")
+    def update_selected_error(self):
+        if self.selected_mv is None:
+            self.selected_error = 'No errors'
+        elif self.selected_mv.error_message == '':
+            self.selected_error = 'No errors'
+        else:
+            self.selected_error = self.selected_mv.error_message
+
+    @on_trait_change("workflow_mv.verify_workflow_event")
+    def received_verify_request(self):
+        """Checks if the root node of workflow tree is requesting a
+        verification of the workflow"""
+        self.verify_workflow_event = True
+
+    def _verify_workflow_event_fired(self):
+        """Verify the workflow and update the modelviews in the workflow tree
+        with any errors"""
+        result = verify_workflow(self.model)
+        parent_child = {WorkflowModelView: ['mco_mv',
+                                            'notification_listeners_mv',
+                                            'execution_layers_mv'],
+                        MCOModelView: ['mco_parameters_mv', 'kpis_mv'],
+                        ExecutionLayerModelView: ['data_sources_mv']}
+        verify_tree(self.workflow_mv, parent_child, result)
+
+
+def verify_tree(mv_list, mappings, errors):
+    """ Call with a list of modelview objects to assign
+    their error messages, plus the error message of
+    their subsequent child modelviews.
+
+    :param mv_list
+    A list of ModelViews
+    :param mappings
+    A dict containing the names of child modelview lists,
+    for each type of modelview
+    :param errors
+    A list of VerifierError, where object.error is an error message and
+    object.subject is the component of the workflow it applies to
+    """
+    if not isinstance(mv_list, list):
+        mv_list = [mv_list]
+
+    for modelview in mv_list:
+        modelview.error_message = ''
+        for type, lists in mappings.items():
+            if isinstance(modelview, type):
+                for child_mv_list_name in lists:
+                    child_mv_list = getattr(modelview, child_mv_list_name)
+                    child_errors = verify_tree(child_mv_list,
+                                               mappings,
+                                               errors)
+                    modelview.error_message += child_errors
+                    if modelview.error_message != '':
+                        modelview.valid = False
+
+    errors_from_list = ''
+
```

Comment:
From here on, you lose me. We need to talk about the rest because I can't understand it.